New take on SI-6534, value classes.

paulp opened the pull request:
Don't prohibit equals and hashCode in universal traits;
instead, always override them in value classes.

paulp committed:
New take on SI-6534, value classes.

Don't prohibit equals and hashCode in universal traits;
instead, always override them in value classes.

paulp commented:
Review by @odersky.

paulp commented:
The warning is behind -Xlint if that's not obvious.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/775/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1486/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1486/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/775/

paulp committed:
Modification to SI-6534 patch.

Only exclude hashCode and equals from being overridden in
value classes, not other synthetics which may turn up such
as case class methods.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/793/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1503/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/793/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1503/
